src/splitpane/SplitPane.ts

```diff
@@ -0,0 +1,190 @@
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { v } from '@dojo/widget-core/d';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+
+import * as css from './styles/splitPane.m.css';
+
+/**
+ * Direction of this SplitPane
+ */
+export const enum Direction {
+	column,
+	row
+};
+
+/**
+ * @type SplitPaneProperties
+ *
+ * Properties that can be set on a SplitPane component
+ *
+ * @property direction      Orientation of this SplitPane, can be `row` or `column`
+ * @property leading        Content to show in the primary pane
+ * @property onResize       Called when the divider is dragged; should be used to update `size`
+ * @property size           Size of the primary pane
+ * @property trailing       Content to show in the secondary pane
+ */
+export interface SplitPaneProperties extends ThemeableProperties {
+	direction?: Direction;
+	leading?: DNode;
+	onResize?(size: number): void;
+	size?: number;
+	trailing?: DNode;
+};
+
+export const SplitPaneBase = ThemeableMixin(WidgetBase);
+
+const DEFAULT_SIZE = 100;
+
+@theme(css)
+export default class SplitPane extends SplitPaneBase<SplitPaneProperties> {
+	private _divider: HTMLElement;
+	private _dragging: boolean;
+	private _lastSize: number;
+	private _position: number;
+	private _root: HTMLElement;
+	private _boundHandlers: any[];
+
+	constructor() {
+		/* istanbul ignore next: disregard transpiled `super`'s "else" block */
+		super();
+
+		/**
+		 * `mouseup` and other events aren't triggered when a user's cursor leaves div.root, so
+		 * global handlers are need to listen to the document instead. SplitPane
+		 * uses a `_dragging` flag so no handlers will be erroneously executed
+		 * if a user isn't actually resizing this SplitPane instance.
+		 */
+		this._boundHandlers = [];
+		[
+			{ event: 'mouseup', func: this._onDragEnd.bind(this) },
+			{ event: 'mousemove', func: this._onDragMove.bind(this) },
+			{ event: 'touchmove', func: this._onDragMove.bind(this) }
+		].forEach(object => {
+			document.addEventListener(object.event, object.func);
+			this._boundHandlers.push(object);
+		});
+
+		this.own(createHandle(() => {
+			this._boundHandlers.forEach(object => document.removeEventListener(object.event, object.func));
+		}));
+
+		const viewportSubscription = observeViewport({
+			next: debounce(this._onResize.bind(this), 200)
+		});
+
+		this.own(createHandle(() => {
+			viewportSubscription.unsubscribe();
+		}));
+	}
+
+	private _getPosition(event: MouseEvent & TouchEvent) {
+		const { direction = Direction.row } = this.properties;
+
+		if (direction === Direction.row) {
+			return event.changedTouches ? event.changedTouches[0].clientX : event.clientX;
+		}
+		else {
+			return event.changedTouches ? event.changedTouches[0].clientY : event.clientY;
+		}
+	}
+
+	private _onDragStart(event: MouseEvent & TouchEvent) {
+		this._dragging = true;
+		this._position = this._getPosition(event);
+	}
```

Comment:
On mobile, without `event.preventDefault` will this result in the scroll position being changed as the divider is dragged?